code/processes/wdb.q

```diff
@@ -506,6 +511,19 @@ clearwdbdata:{[]
 		.lg.o[`deletewdbdata;"no directory found at ",1_string wdbpart]		
 	];
 	};
+
+/ - if there is data in the tailDB directory for the partition remove it before replay
+/ - is only run during datastriping mode
+cleartaildir:{
+  $[saveenabled and not () ~ key ` sv(.ds.td;.proc.procname;`$string .wdb.currentpartition);
+   [.lg.o[`deletetaildb;"removing taildb (",(delstrg:1_string ` sv(.ds.td;.proc.procname;`$string .wdb.currentpartition)),") prior to log replay"];
+   @[.os.deldir;delstrg;{[e] .lg.e[`deletewdbdata;"Failed to delete existing taildir data.  Error was : ",e];'e }];
+    .lg.o[`deletewdbdata;"finished removing taildb data prior to log replay"];
+    ];
+    .lg.o[`deletewdbdata;"no directory found at ",1_string ` sv(.ds.td;.proc.procname;`$string .wdb.currentpartition)]
+  ];
```

Comment:
IMO this conditional is a little hard to follow (hard to see where the true branch ends etc.) - I would reverse the condition and use an `if` with an early exit, something like

```suggestion
  if[not saveenabled and not () ~ key ` sv(.ds.td;.proc.procname;`$string .wdb.currentpartition);
    .lg.o[`deletewdbdata;"no directory found at ",1_string ` sv(.ds.td;.proc.procname;`$string .wdb.currentpartition)];
    :();
  ];
  .lg.o[`deletetaildb;"removing taildb (",(delstrg:1_string ` sv(.ds.td;.proc.procname;`$string .wdb.currentpartition)),") prior to log replay"];
  @[.os.deldir;delstrg;{[e] .lg.e[`deletewdbdata;"Failed to delete existing taildir data.  Error was : ",e];'e }];
  .lg.o[`deletewdbdata;"finished removing taildb data prior to log replay"];
```

Reply:
I am slightly confused about this I think. I wanted two cases:
1. If it is running as a tailer (in savedown mode and in datastriping mode) and a tailDB exists, it should delete the db.
2. If not, just log output that there is no taildb/savedown mode is off.

I think the changes would make the os delete the taildb even if it doesn't exist, but could be wrong.

Reply:
In my suggested change, the `:();` at the end of the `if` will return early from the function if the conditional is entered, so nothing after the `if` conditional will be executed in that case

Reply:
Ah right, thanks for the help can change it to that then